.github/workflows/build_and_test.yml

```diff
@@ -472,6 +472,7 @@ jobs:
           CHECK_ONDISK_DATA_COMPATIBILITY: nonempty
           BUILD_TAG: ${{ needs.tag.outputs.build-tag }}
           PAGESERVER_VIRTUAL_FILE_IO_ENGINE: std-fs
+          PAGESERVER_GET_VECTORED_IMPL: vectored
```

Comment:
Good point. I was trying to avoid the parametrisation, but looks like it's needed.

Reply:
Well it's not really needed because you probably don't want to 2x the number of CI jobs. But, you could have it it while this PR is under development and revert shortly before merging.

It's the same considerations / trade-off that I have with shepherding tokio-epoll-uring into production. Let's discuss the strategy with the team in the next planning meeting.

Reply:
We discussed this out of band. I've enabled validation for debug *and* release modes. This has no production/staging impact since vectored get is not enabled by default there. It also means that we implicitly regress test the sequential impl currently used in prod.

For releasing, we will most likely on the validation in debug mode.